x/sockopt/is_sending_bytes_linux.go

```diff
@@ -0,0 +1,42 @@
+//go:build linux
+
+package sockopt
+
+import (
+	"net"
+
+	"golang.org/x/sys/unix"
+)
+
+func isSocketFdSendingBytes(fd int) (bool, error) {
+	tcpInfo, err := unix.GetsockoptTCPInfo(int(fd), unix.IPPROTO_TCP, unix.TCP_INFO)
+	if err != nil {
+		return false, err
+	}
+
+	// 1 == TCP_ESTABLISHED, but for some reason not available in the package
+	if tcpInfo.State != 1 {
+		// If the connection is not established, the socket is not sending bytes
+		return false, nil
+	}
```

Comment:
could you confirm this does what we supposed the bye-dpi code does? I'm starting having some doubts looking [at the list of states](https://elixir.bootlin.com/linux/v6.10.6/source/include/net/tcp_states.h)

Reply:
https://github.com/hufrea/byedpi/blob/main/desync.c#L99